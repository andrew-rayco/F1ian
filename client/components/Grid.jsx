import React from 'react'

import * as api from '../api'
import RaceOptions from './RaceOptions'

export default class Grid extends React.Component {
  constructor() {
    super(),
    this.state = {}
  }

  componentWillMount() {
    let location = this.props.location.pathname
    let pathArray = location.split('/')
    let season = pathArray[2]
    let raceId = pathArray[3]
    api.getGrid(season, raceId, (grid) => {
      this.setState({grid})
    })
  }

  listGrid(fullGrid) {
    return fullGrid.map((driver) => {
      return (
        <tr key={driver.resultId}>
          <td><strong>{driver.grid}</strong></td>
          <td><a href={driver['driver-url']}>{driver.forename} {driver.surname}</a></td>
          <td><a href={driver.constructorUrl}>{driver.constructorName}</a></td>
        </tr>
      )
    })
  }

  render() {
    if (this.state.grid) {
      let grid = this.state.grid.gridData
      console.log(grid)
      return (
        <div className="grid">
          <h2>{grid[0].year} {grid[0].raceName}</h2>
          <h3>Starting Grid</h3>
          <table>
            <thead>
              <tr>
                <th>Position</th>
                <th>Driver</th>
                <th>Team</th>
              </tr>
            </thead>
            <tbody>
              {this.listGrid(grid)}
            </tbody>
          </table>
          <RaceOptions props={grid[0]}/>
        </div>
      )
    } else {
      return <div></div>
    }
  }

}
