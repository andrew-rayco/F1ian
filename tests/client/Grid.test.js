import React from 'react'
import { shallow } from 'enzyme'

import Grid from '../../client/components/Grid'

describe('Grid component', () => {

    const location = {
        pathname: '0/1/2015/2'
    }

    const wrapper = shallow(<Grid location={ location } />)

    wrapper.setState({
        grid: {
            gridData: [{
                year: 1985,
                raceName: 'Australian Grand Prix',
                resultId: 9998
            }, {
                year: 1986,
                raceName: 'Brazilian Grand Prix',
                resultId: 9999
            }]
        }
    })

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('renders returned data', () => {
        expect(wrapper.find('tr').length).toEqual(3)
    })

    it('renders Loading component if no data', () => {
        wrapper.setState({ grid: null })
        expect(wrapper.text()).toMatch(/<Loading \/>/)
    })

    it('uses props if passed props', () => {
        const wrapperWithProps = shallow(
            <Grid
                location={ location }
                season={'2015'}
                raceId={2}
                round={2}
            />)
        // what to test here?
        // fail test on purpose to pick up where I left off
        expect(3).toEqual(2)
    })
})
