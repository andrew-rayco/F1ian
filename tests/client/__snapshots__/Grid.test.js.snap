// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Grid
    location={
      Object {
        "pathname": "0/1/2015/999",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          1985
           
          Australian Grand Prix
        </h2>,
        <h3>
          Starting Grid
        </h3>,
        <table>
          <thead>
            <tr>
              <th>
                Position
              </th>
              <th>
                Driver
              </th>
              <th>
                Team
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                <strong />
              </td>
              <td>
                <a
                  href={undefined}
                >
                   
                </a>
              </td>
              <td>
                <a
                  href={undefined}
                />
              </td>
            </tr>
            <tr>
              <td>
                <strong />
              </td>
              <td>
                <a
                  href={undefined}
                >
                   
                </a>
              </td>
              <td>
                <a
                  href={undefined}
                />
              </td>
            </tr>
          </tbody>
        </table>,
      ],
      "className": "grid sub-section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            1985,
            " ",
            "Australian Grand Prix",
          ],
        },
        "ref": null,
        "rendered": Array [
          1985,
          " ",
          "Australian Grand Prix",
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Starting Grid",
        },
        "ref": null,
        "rendered": "Starting Grid",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <thead>
              <tr>
                <th>
                  Position
                </th>
                <th>
                  Driver
                </th>
                <th>
                  Team
                </th>
              </tr>
            </thead>,
            <tbody>
              <tr>
                <td>
                  <strong />
                </td>
                <td>
                  <a
                    href={undefined}
                  >
                     
                  </a>
                </td>
                <td>
                  <a
                    href={undefined}
                  />
                </td>
              </tr>
              <tr>
                <td>
                  <strong />
                </td>
                <td>
                  <a
                    href={undefined}
                  >
                     
                  </a>
                </td>
                <td>
                  <a
                    href={undefined}
                  />
                </td>
              </tr>
            </tbody>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <tr>
                <th>
                  Position
                </th>
                <th>
                  Driver
                </th>
                <th>
                  Team
                </th>
              </tr>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <th>
                    Position
                  </th>,
                  <th>
                    Driver
                  </th>,
                  <th>
                    Team
                  </th>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Position",
                  },
                  "ref": null,
                  "rendered": "Position",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Driver",
                  },
                  "ref": null,
                  "rendered": "Driver",
                  "type": "th",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Team",
                  },
                  "ref": null,
                  "rendered": "Team",
                  "type": "th",
                },
              ],
              "type": "tr",
            },
            "type": "thead",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <tr>
                  <td>
                    <strong />
                  </td>
                  <td>
                    <a
                      href={undefined}
                    >
                       
                    </a>
                  </td>
                  <td>
                    <a
                      href={undefined}
                    />
                  </td>
                </tr>,
                <tr>
                  <td>
                    <strong />
                  </td>
                  <td>
                    <a
                      href={undefined}
                    >
                       
                    </a>
                  </td>
                  <td>
                    <a
                      href={undefined}
                    />
                  </td>
                </tr>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <td>
                      <strong />
                    </td>,
                    <td>
                      <a
                        href={undefined}
                      >
                         
                      </a>
                    </td>,
                    <td>
                      <a
                        href={undefined}
                      />
                    </td>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <strong />,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": undefined,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "strong",
                    },
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <a
                        href={undefined}
                      >
                         
                      </a>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          undefined,
                          " ",
                          undefined,
                        ],
                        "href": undefined,
                      },
                      "ref": null,
                      "rendered": Array [
                        undefined,
                        " ",
                        undefined,
                      ],
                      "type": "a",
                    },
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <a
                        href={undefined}
                      />,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": undefined,
                        "href": undefined,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "a",
                    },
                    "type": "td",
                  },
                ],
                "type": "tr",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <td>
                      <strong />
                    </td>,
                    <td>
                      <a
                        href={undefined}
                      >
                         
                      </a>
                    </td>,
                    <td>
                      <a
                        href={undefined}
                      />
                    </td>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <strong />,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": undefined,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "strong",
                    },
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <a
                        href={undefined}
                      >
                         
                      </a>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          undefined,
                          " ",
                          undefined,
                        ],
                        "href": undefined,
                      },
                      "ref": null,
                      "rendered": Array [
                        undefined,
                        " ",
                        undefined,
                      ],
                      "type": "a",
                    },
                    "type": "td",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <a
                        href={undefined}
                      />,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": undefined,
                        "href": undefined,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "a",
                    },
                    "type": "td",
                  },
                ],
                "type": "tr",
              },
            ],
            "type": "tbody",
          },
        ],
        "type": "table",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            1985
             
            Australian Grand Prix
          </h2>,
          <h3>
            Starting Grid
          </h3>,
          <table>
            <thead>
              <tr>
                <th>
                  Position
                </th>
                <th>
                  Driver
                </th>
                <th>
                  Team
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  <strong />
                </td>
                <td>
                  <a
                    href={undefined}
                  >
                     
                  </a>
                </td>
                <td>
                  <a
                    href={undefined}
                  />
                </td>
              </tr>
              <tr>
                <td>
                  <strong />
                </td>
                <td>
                  <a
                    href={undefined}
                  >
                     
                  </a>
                </td>
                <td>
                  <a
                    href={undefined}
                  />
                </td>
              </tr>
            </tbody>
          </table>,
        ],
        "className": "grid sub-section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              1985,
              " ",
              "Australian Grand Prix",
            ],
          },
          "ref": null,
          "rendered": Array [
            1985,
            " ",
            "Australian Grand Prix",
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Starting Grid",
          },
          "ref": null,
          "rendered": "Starting Grid",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <thead>
                <tr>
                  <th>
                    Position
                  </th>
                  <th>
                    Driver
                  </th>
                  <th>
                    Team
                  </th>
                </tr>
              </thead>,
              <tbody>
                <tr>
                  <td>
                    <strong />
                  </td>
                  <td>
                    <a
                      href={undefined}
                    >
                       
                    </a>
                  </td>
                  <td>
                    <a
                      href={undefined}
                    />
                  </td>
                </tr>
                <tr>
                  <td>
                    <strong />
                  </td>
                  <td>
                    <a
                      href={undefined}
                    >
                       
                    </a>
                  </td>
                  <td>
                    <a
                      href={undefined}
                    />
                  </td>
                </tr>
              </tbody>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <tr>
                  <th>
                    Position
                  </th>
                  <th>
                    Driver
                  </th>
                  <th>
                    Team
                  </th>
                </tr>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <th>
                      Position
                    </th>,
                    <th>
                      Driver
                    </th>,
                    <th>
                      Team
                    </th>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Position",
                    },
                    "ref": null,
                    "rendered": "Position",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Driver",
                    },
                    "ref": null,
                    "rendered": "Driver",
                    "type": "th",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Team",
                    },
                    "ref": null,
                    "rendered": "Team",
                    "type": "th",
                  },
                ],
                "type": "tr",
              },
              "type": "thead",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <tr>
                    <td>
                      <strong />
                    </td>
                    <td>
                      <a
                        href={undefined}
                      >
                         
                      </a>
                    </td>
                    <td>
                      <a
                        href={undefined}
                      />
                    </td>
                  </tr>,
                  <tr>
                    <td>
                      <strong />
                    </td>
                    <td>
                      <a
                        href={undefined}
                      >
                         
                      </a>
                    </td>
                    <td>
                      <a
                        href={undefined}
                      />
                    </td>
                  </tr>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <td>
                        <strong />
                      </td>,
                      <td>
                        <a
                          href={undefined}
                        >
                           
                        </a>
                      </td>,
                      <td>
                        <a
                          href={undefined}
                        />
                      </td>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <strong />,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": undefined,
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "strong",
                      },
                      "type": "td",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <a
                          href={undefined}
                        >
                           
                        </a>,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            undefined,
                            " ",
                            undefined,
                          ],
                          "href": undefined,
                        },
                        "ref": null,
                        "rendered": Array [
                          undefined,
                          " ",
                          undefined,
                        ],
                        "type": "a",
                      },
                      "type": "td",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <a
                          href={undefined}
                        />,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": undefined,
                          "href": undefined,
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "a",
                      },
                      "type": "td",
                    },
                  ],
                  "type": "tr",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <td>
                        <strong />
                      </td>,
                      <td>
                        <a
                          href={undefined}
                        >
                           
                        </a>
                      </td>,
                      <td>
                        <a
                          href={undefined}
                        />
                      </td>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <strong />,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": undefined,
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "strong",
                      },
                      "type": "td",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <a
                          href={undefined}
                        >
                           
                        </a>,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            undefined,
                            " ",
                            undefined,
                          ],
                          "href": undefined,
                        },
                        "ref": null,
                        "rendered": Array [
                          undefined,
                          " ",
                          undefined,
                        ],
                        "type": "a",
                      },
                      "type": "td",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <a
                          href={undefined}
                        />,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": undefined,
                          "href": undefined,
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "a",
                      },
                      "type": "td",
                    },
                  ],
                  "type": "tr",
                },
              ],
              "type": "tbody",
            },
          ],
          "type": "table",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
